### DISCUSSION TITLE
Don't read /etc/hosts

### DISCUSSION THREAD

**Discussion by alexdelprete** [AdguardTeam/AdGuardHome#4125] [Q&A]
Is there a way to force AGH not to use /etc/hosts? I read about the `--no-etc-hosts ` option, so I restarted the service with that, but it didn't seem to have effect. Does it only apply during the install phase? It would be good to have that as an option in the UI/config file.

The problem I'm having is that if I do an nslookup on the LAN for the adguardhome.mydomain.lan host (AGH itself), it returns 127.0.1.1 instead of forwarding the request to my opnsense router (which does DHCP and runs unbound).

So I can't resolve the agh hostname on the LAN. For now, I configured a rewrite to bypass the problem, but I don't like it as a permanent solution.

Thanks for any advice.

**Comment by ainar-g** [comment]
Hello.  The `--no-etc-hosts` option is exactly what you'd want.  How did you install AGH?  If you installed it as a service, are you sure that your OS reread the configuration?  (I.e. calling `systemctl daemon-reload` after changing a systemd `.service` file.)

**Reply by alexdelprete** [reply]
Thanks, I tried again and I understood the problem: the format of the parameters passed to the exec. Now it worked fine.

It would be good to have that in the options, configurable through the UI though.
